src/core/service.c

```diff
@@ -2140,6 +2141,83 @@ _pure_ static bool service_can_reload(Unit *u) {
         return !!s->exec_command[SERVICE_EXEC_RELOAD];
 }
 
+static unsigned service_exec_command_index(Unit *u, ServiceExecCommand id, bool control) {
```

Comment:
i think it would be nicer to drop the "control" argument here, and instead simply take an ExecCommand* parameter, which is then either passed as `s->control_command` or `s->main_command`.